analyzer/metadata-plugin/src/main/java/eu/fasten/analyzer/metadataplugin/MetadataDatabasePlugin.java

```diff
@@ -58,30 +59,35 @@ public MetadataDatabasePlugin(PluginWrapper wrapper) {
         private boolean restartTransaction = false;
         private final int transactionRestartLimit = 3;
 
-        /**
-         * Constructor for MetadataPlugin with default DSLContext
-         * with parameters from `postgres.properties`.
-         *
-         * @param callgraphTopic Topic from which to consume call graphs
-         * @throws IOException              if cannot read file `postgres.properties`
-         * @throws SQLException             if cannot connect to the database
-         * @throws IllegalArgumentException if database URL in `postgres.properties` is malformed
-         */
-        public MetadataDBExtension(String callgraphTopic) throws IOException, SQLException,
-                IllegalArgumentException {
-            this(callgraphTopic, PostgresConnector.getDSLContext());
-        }
+//        /**
+//         * Constructor for MetadataPlugin with default DSLContext
+//         * with parameters from `postgres.properties`.
+//         *
+//         * @param callgraphTopic Topic from which to consume call graphs
+//         * @throws IOException              if cannot read file `postgres.properties`
+//         * @throws SQLException             if cannot connect to the database
+//         * @throws IllegalArgumentException if database URL in `postgres.properties` is malformed
+//         */
+//        public MetadataDBExtension(String callgraphTopic) throws IOException, SQLException,
+//                IllegalArgumentException {
+//            this(callgraphTopic, PostgresConnector.getDSLContext());
+//        }
 
-        /**
-         * Constructor for MetadataPlugin with provided DSLContext.
-         *
-         * @param callgraphTopic Topic from which to consume call graphs
-         * @param dslContext     DSLContext for jOOQ to query the database
-         */
-        public MetadataDBExtension(String callgraphTopic, DSLContext dslContext) {
-            super();
-            this.dslContext = dslContext;
-            this.topic = callgraphTopic;
+//        /**
+//         * Constructor for MetadataPlugin with provided DSLContext.
+//         *
+//         * @param callgraphTopic Topic from which to consume call graphs
+//         * @param dslContext     DSLContext for jOOQ to query the database
+//         */
+//        public MetadataDBExtension(String callgraphTopic, DSLContext dslContext) {
+//            super();
+//            this.dslContext = dslContext;
+//            this.topic = callgraphTopic;
+//        }
+
+        @Override
+        public void getDBAccess(String DBUrl, String username, String password) throws SQLException {
```

Comment:
I think this method should also throw `IllegalArgumentException` since in `PostgresConnector.getDSLContext(...)` there is a check if DBUrl is in correct format and if it is not, it will throw `IllegalArgumentException`

Reply:
I think it might be a better idea to catch `IllegalArgumentException` in the `getDBAccess` and set plug-in error rather than catching it at higher levels like the Main class.
Moreover, the project compiles successfully without throwing `IllegalArgumentException` in the `getDBAccess` and the IntelliJ also doesn't complain either!

Reply:
It does not throw it because the URL we use is valid. But if user inputs invalid URL as a command-line argument it will throw uncaught `IllegalArgumentException`.
However, catching it in the `getDBAccess(..)` seems like a good solution too.